File: app/src/main/java/com/musicplayer/OpenMusic/data/base/FileConverter.java
package com.musicplayer.OpenMusic.data.base;

import androidx.room.TypeConverter;

import java.io.File;

public class FileConverter {

    @TypeConverter
    public static String fromFile(File file) {
        return file.getAbsolutePath();
    }

    @TypeConverter
    public static File fileFromString(String path) {
        return new File(path);
    }
}

File: app/src/main/kotlin/com/musicplayer/openmusic/data/base/FileConverter.kt
package com.musicplayer.openmusic.data.base

import androidx.room.TypeConverter
import java.io.File

object FileConverter {
    @JvmStatic
    @TypeConverter
    fun fromFile(file: File): String {
        return file.absolutePath
    }

    @JvmStatic
    @TypeConverter
    fun fileFromString(path: String): File {
        return File(path)
    }
}